src/Bicep.Core/Emit/InlineDependencyVisitor.cs

```diff
@@ -146,6 +146,16 @@ private void VisitVariableAccessSyntaxInternal(VariableAccessSyntax syntax)
 
             switch (model.GetSymbolInfo(syntax))
             {
+                case ParameterSymbol parameterSymbol:
+                    // The the parameter is a resource type this call needs to be inlined
+                    // because references cannot be used in variables.
+                    if (parameterSymbol.DeclaringParameter.Type is ResourceTypeSyntax)
+                    {
+                        SetInlineCache(true);
+                    }
+                    
+                    return;
```

Comment:
Instead of adding a new case here, I think it would be better to add a new match to the last block (the one that starts on line 189):

```csharp
case ResourceSymbol:
case ModuleSymbol:
case ParameterSymbol parameterSymbol when model.ResourceMetadata.TryLookup(syntax) is ResourceMetadata:
    if (this.currentDeclaration is not null && shouldInlineCache[currentDeclaration] != Decision.SkipInline)
```

`model.ResourceMetadata` will determine whether the parameter is resource-typed, and the logic handling `ResourceSymbol`s and `ModuleSymbol`s has a guard to make sure deploy-time constant properties (like `<resource>.id`, `<resource>.apiVersion`, etc) aren;t unnecessarily inlined.

Reply:
This is done